desc/equilibrium/utils.py

```diff
@@ -182,3 +184,124 @@ def parse_axis(axis, NFP=1, sym=True, surface=None):
+    copy : bool
+        Whether or not to return a copy or to modify the original equilibrium.
+
+    Returns
+    -------
+    eq_inner: New Equilibrium object, contracted from the old one such that
+        eq.pressure(rho=inner_rho) = eq_inner.pressure(rho=1), and
+        eq_inner LCFS = eq's rho=inner_rho surface.
+        Note that this will not be in force balance, and so must be re-solved.
+
+    """
+    errorif(
+        not (inner_rho < 1 and inner_rho > 0),
+        ValueError,
+        f"Surface must be in the range 0 < inner_rho < 1, instead got {inner_rho}.",
+    )
+
+    def scale_profile(profile, rho):
+        if profile is None:
+            return profile
+        is_power_series = isinstance(profile, PowerSeriesProfile)
+        if contract_profiles and is_power_series:
+            # only PowerSeriesProfile both
+            # a) has a from_values
+            # b) can safely use that from_values to represent a
+            #    subset of itself.
+            x = np.linspace(0, 1, eq.L_grid)
+            grid = LinearGrid(rho=x / rho)
+            y = profile.compute(grid)
+            return profile.from_values(x=x, y=y)
+        elif contract_profiles:
+            warnif(
+                not isinstance(profile, SplineProfile),
+                UserWarning,
+                f"{profile} is not a PowerSeriesProfile or SplineProfile,"
+                " so cannot safely contract using the same profile type."
+                "falling back to fitting the values with a SplineProfile",
+            )
+            x = np.linspace(0, 1, eq.L_grid)
+            grid = LinearGrid(rho=x / rho)
+            y = profile.compute(grid)
+            return SplineProfile(knots=x, values=y)
+        else:  # don't do any scaling of the profile
+            return profile
+
+    # create new profiles for contracted equilibrium
+    pressure = scale_profile(eq.pressure, inner_rho)
+    iota = scale_profile(eq.iota, inner_rho)
+    current = scale_profile(eq.current, inner_rho)
+    electron_density = scale_profile(eq.electron_density, inner_rho)
+    electron_temperature = scale_profile(eq.electron_temperature, inner_rho)
+    ion_temperature = scale_profile(eq.ion_temperature, inner_rho)
+    atomic_number = scale_profile(eq.atomic_number, inner_rho)
+    anisotropy = scale_profile(eq.anisotropy, inner_rho)
+
+    surf_inner = eq.get_surface_at(rho=inner_rho)
+    surf_inner.rho = 1.0
+    from .equilibrium import Equilibrium
+
+    eq_inner = Equilibrium(
+        surface=surf_inner,
+        pressure=pressure,
+        iota=iota,
+        current=current,
+        electron_density=electron_density,
+        electron_temperature=electron_temperature,
+        ion_temperature=ion_temperature,
+        atomic_number=atomic_number,
+        anisotropy=anisotropy,
+        Psi=float(
+            eq.compute("Psi", grid=LinearGrid(rho=inner_rho, NFP=eq.NFP))["Psi"][0]
+        ),  # flux (in Webers) within the new last closed flux surface
```

Comment:
Technically we don't need to include NFP here, since the compute function for `"Psi"` is just $\Psi = \Psi_0 \rho^2$. So for simplicity we could use `grid=Grid([inner_rho, 0, 0])`